cmd/sales-api/handlers/user.go

```diff
@@ -107,10 +116,33 @@ func (u *User) Delete(ctx context.Context, log *log.Logger, w http.ResponseWrite
 	defer dbConn.Close()
 
 	err := user.Delete(ctx, dbConn, params["id"])
-	if err = check(err); err != nil {
+	if err = translate(err); err != nil {
 		return errors.Wrapf(err, "Id: %s", params["id"])
 	}
 
 	web.Respond(ctx, log, w, nil, http.StatusNoContent)
 	return nil
 }
+
+// Token handles a request to authenticate a user. It expects a request using
+// Basic Auth with a user's email and password. It responds with a JWT.
+func (u *User) Token(ctx context.Context, log *log.Logger, w http.ResponseWriter, r *http.Request, params map[string]string) error {
+	ctx, span := trace.StartSpan(ctx, "handlers.User.Token")
+	defer span.End()
+
+	dbConn := u.MasterDB.Copy()
+	defer dbConn.Close()
+
+	email, pass, ok := r.BasicAuth()
+	if !ok {
+		return web.ErrUnauthorized
+	}
+
+	tkn, err := user.Authenticate(ctx, dbConn, u.Auth.Key, u.Auth.KeyID, u.Auth.Alg, email, pass)
```

Comment:
The name `Authenticate` for the func was confusing to me due to above comment

Reply:
Hmm I would be open to other names if you have suggestions? The signature of the function includes `email` and `pass` for what it's worth.

Reply:
Good point, might have had token auth on my mind

Reply:
I would prefer something like `GetToken` but I may be nitpicking here